airbyte-ci/connectors/pipelines/pipelines/airbyte_ci/connectors/test/steps/common.py

```diff
@@ -153,8 +153,44 @@ def have_same_major_minor_patch(self) -> bool:
             and self.master_connector_version.patch == self.current_connector_version.patch
         )
 
+    @staticmethod
+    def _get_registry_override_tag(metadata: Optional[dict], channel: str) -> Optional[str]:
+        """Extract the dockerImageTag from registryOverrides for a given channel.
+
+        Args:
+            metadata: The metadata dictionary (master or current).
+            channel: The channel to extract from ("cloud" or "oss").
+
+        Returns:
+            The dockerImageTag value if present, None otherwise.
+        """
+        if metadata is None:
+            return None
+        try:
+            return metadata.get("data", {}).get("registryOverrides", {}).get(channel, {}).get("dockerImageTag")
+        except (TypeError, AttributeError):
+            return None
+
+    def _has_registry_override_docker_tag_change(self) -> bool:
+        """Check if registryOverrides.cloud.dockerImageTag or registryOverrides.oss.dockerImageTag has changed.
+
+        Returns:
+            bool: True if either cloud or oss dockerImageTag has been added, removed, or changed.
+        """
+        master_cloud = self._get_registry_override_tag(self.master_metadata, "cloud")
+        current_cloud = self._get_registry_override_tag(self.context.metadata, "cloud")
+
+        master_oss = self._get_registry_override_tag(self.master_metadata, "oss")
+        current_oss = self._get_registry_override_tag(self.context.metadata, "oss")
+
+        return (master_cloud != current_cloud) or (master_oss != current_oss)
+
     def validate(self) -> StepResult:
         if self.is_version_not_incremented():
+            # Allow version to stay the same if registryOverrides.cloud.dockerImageTag or
+            # registryOverrides.oss.dockerImageTag has been added, removed, or changed
+            if self._has_registry_override_docker_tag_change():
```

Comment:
can you also make the same registry override validation change for the `airbyte-ci/connectors/connectors_qa/src/connectors_qa/checks/version.py` which is also used during CI